src/components/ApiAction/QueryCard.vue

```diff
@@ -0,0 +1,353 @@
+                  target="_blank"
+                  >API Documentation
+                  <i class="fa fa-external-link-alt" />
+                </a>
+                .
+              </b-popover>
+            </b-row>
+          </b-col>
+          <b-col class="text-right" id="query-button-actions" cols="4">
+            <b-tooltip
+              v-if="!isQueryValid(jsonQuery)"
+              target="query-button-actions"
+              placement="bottom"
+            >
+              The query is invalid.
+            </b-tooltip>
+
+            <b-button
+              @click="performQuery"
+              :disabled="!isQueryValid(jsonQuery)"
+              variant="success"
+              :data-cy="`api-actions-run-button-${tabIdx}`"
+              class="mr-3 pointer"
+            >
+              <i class="fas fa-rocket mr-2" />
+              RUN
+            </b-button>
+            <b-button
+              @click="saveQuery"
+              :disabled="!isQueryValid(jsonQuery)"
+              :data-cy="`api-actions-save-button-${tabIdx}`"
+              variant="outline-primary"
+              class="pointer"
+            >
+              <i class="fas fa-save mr-2" />
+              SAVE
+            </b-button>
+          </b-col>
+        </b-row>
+        <b-row align-v="stretch" class="multipaneRow px-3">
+          <Multipane
+            class="QueryLayout-vertical Query-Custom-resizer-vertical"
+            layout="vertical"
+          >
+            <div
+              class="QueryLayout-sidebarWrapper-vertical"
+              data-cy="QueryLayout-sidebarWrapper"
+            >
+              <b-card no-body class="h-100">
+                <json-editor
+                  :id="`queryEditorWrapper-${tabIdx}`"
+                  :ref="`queryEditorWrapper-${tabIdx}`"
+                  class="m-2 h-100"
+                  :data-cy="`api-actions-query-JSONEditor-${tabIdx}`"
+                  :content="jsonQuery"
+                  @change="queryBodyChange"
+                />
+              </b-card>
+            </div>
+            <MultipaneResizer data-cy="sidebarResizer" />
+            <div class="QueryLayout-contentWrapper-vertical">
+              <ResponseCard :tabIdx="tabIdx" :response="response" />
+            </div>
+          </Multipane>
+        </b-row>
+      </b-container>
+    </b-card-text>
+  </b-card-body>
+</template>
+
+<script>
+import jsonEditor from '@/components/Common/JsonEditor'
+import ResponseCard from '@/components/ApiAction/ResponseCard'
+import { Multipane, MultipaneResizer } from 'vue-multipane'
+import _ from 'lodash'
+
+export default {
+  name: 'QueryCard',
+  props: {
+    query: {},
+    tabIdx: {},
+    controllers: {},
+    actions: {},
+    api: {},
+    openapi: {},
+    response: {}
+  },
+  components: {
+    jsonEditor,
+    ResponseCard,
+    Multipane,
+    MultipaneResizer
+  },
+  data() {
+    return {
+      isFullScreen: false,
+      jsonQuery: '{}',
+      editedQuery: {
```

Comment:
Edited Query === Current Query ?

Reply:
There is no `currentQuery` in `QueryCard` scope
The `editedQuery` is the one which is updated using both jsonEditor and inputs, the changes are emitted to the parent component to keep synchronized the queries list of the parent (and also the `query` props and `editedQuery` data of `QueryCard` component.
The parent component (`ApiAction`) only have a `currentQueryName` (and no `currentQuery` ) which is used to set (or not) the active entry of the `QueryList`

Reply:
It's a bit confusing...
If you have any naming proposal do not hesitate :)

Reply:
Another point to talk about on Discord :)